sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/XXE/XmlReaderSettingsValidator.cs

```diff
@@ -28,39 +29,54 @@
 
 namespace SonarAnalyzer.Rules.XXE
 {
+    /// <summary>
+    /// This class is responsible to check if a XmlReaderSettings node is vulnerable to XXE attacks.
+    ///
+    /// By default the class is safe:
+    ///  - before .Net 4.5.2 it has XmlResolver set to an instance of XmlUrlResolver which is not secure but since has ProhibitDtd set to false it is safe to use
+    ///  - starting with .Net 4.5.2 XmlResolver is set to null, ProhibitDtd (now obsolete) set to false and DtdProcessing set to ignore
+    ///
+    /// If the properties are modified, in order to be secure, we have to check that either ProhibitDtd is set to false, DtdProcessing set to Ignore or
+    /// XmlResolver is set to null or is different than XmlUrlResolver and XmlPreloadedResolver.
+    /// </summary>
     internal class XmlReaderSettingsValidator
     {
         private readonly SemanticModel semanticModel;
+        private readonly bool isXmlResolverSafeByDefault;
 
-        public XmlReaderSettingsValidator(SemanticModel semanticModel)
+        public XmlReaderSettingsValidator(SemanticModel semanticModel, NetFrameworkVersion version)
         {
             this.semanticModel = semanticModel;
+            this.isXmlResolverSafeByDefault = IsXmlResolverPropertySafeByDefault(version);
         }
 
-        public bool IsUnsafe(InvocationExpressionSyntax invocation, ISymbol settings, NetFrameworkVersion version)
+        /// <summary>
+        /// Checks if a method invocation (e.g. XmlReader.Create) receives a secure instance of XmlReaderSettings.
+        /// </summary>
+        /// <param name="invocation">A method invocation syntax node (e.g. XmlReader.Create).</param>
+        /// <param name="settings">The symbol of the XmlReaderSettings node received as parameter. This is used to check
+        /// if certain properties (ProhibitDtd, DtdProcessing or XmlUrlResolver) were modified for the given symbol.</param>
+        /// <returns>true, if the XmlReaderSettings is secure; otherwise false.</returns>
+        public bool IsUnsafe(InvocationExpressionSyntax invocation, ISymbol settings)
         {
             var unsafeDtdProcessing = false;
```

Comment:
I would add a comment above `unsafeDtdProcessing` to say:
```
// By default ProhibitDtd is 'true' and DtdProcessing is 'ignore'
```
There is no risk that the default values will change, so the comment will likely just stay like.